app/uk/gov/hmrc/gform/views/summary/TextFormatter.scala

```diff
@@ -16,42 +16,83 @@
 
 package uk.gov.hmrc.gform.views.summary
 
+import java.text.NumberFormat
+
 import cats.syntax.option._
 import play.api.i18n.Messages
-import uk.gov.hmrc.gform.commons.BigDecimalUtil.toBigDecimalSafe
+import uk.gov.hmrc.gform.commons.BigDecimalUtil._
 import uk.gov.hmrc.gform.eval.smartstring.SmartStringEvaluator
+import uk.gov.hmrc.gform.eval.smartstring.SmartStringEvaluationSyntax
 import uk.gov.hmrc.gform.fileupload.Envelope
 import uk.gov.hmrc.gform.models.Atom
 import uk.gov.hmrc.gform.models.helpers.DateHelperFunctions
-import uk.gov.hmrc.gform.sharedmodel.LocalisedString
+import uk.gov.hmrc.gform.sharedmodel.{ LangADT, LocalisedString, SmartString }
 import uk.gov.hmrc.gform.sharedmodel.formtemplate._
 import uk.gov.hmrc.gform.validation.{ FormFieldValidationResult, HtmlFieldId }
 import uk.gov.hmrc.gform.commons.NumberFormatUtil._
-import uk.gov.hmrc.gform.sharedmodel.LangADT
 
 object TextFormatter {
 
   def componentTextReadonly(currentValue: String, textConstraint: TextConstraint)(implicit l: LangADT): String =
     textConstraint match {
-      case PositiveNumber(_, fracDigits, rm, unit) => formatNumber(currentValue, fracDigits, rm, unit)
-      case Number(_, fracDigits, rm, unit)         => formatNumber(currentValue, fracDigits, rm, unit)
-      case s: Sterling                             => formatSterling(currentValue)
-      case _                                       => currentValue
+      // format: off
+      case IsPositiveNumberOrNumber(PositiveNumberOrNumber(_, maxFractionalDigits, roundingMode, unit)) => formatNumber(currentValue, maxFractionalDigits, roundingMode, unit)
+      case _: Sterling                                                                                  => formatSterling(currentValue)
+      case _                                                                                            => currentValue
+      // format: on
     }
 
-  def componentTextEditable(currentValue: String, textConstraint: TextConstraint): String =
-    textConstraint match {
-      case PositiveNumber(_, _, _, unit) => stripTrailingZeros(currentValue)
-      case Number(_, _, _, unit)         => stripTrailingZeros(currentValue)
-      case s: Sterling                   => stripTrailingZeros(currentValue)
-      case _                             => currentValue
+  def componentTextForRendering(
+    currentValue: String,
+    textConstraint: TextConstraint,
+    presentationHint: Option[List[PresentationHint]],
+    editable: Boolean): String =
+    (textConstraint, presentationHint, editable) match {
+      // format: off
+      case (IsPositiveNumberOrNumber(PositiveNumberOrNumber(_, _, _, _)), _, _) => stripTrailingZeros(currentValue)
```

Comment:
When you don't need anything from `IsPositiveNumberOrNumber` than this is shorter:
```suggestion
      case (IsPositiveNumberOrNumber(_), _, _)                   => stripTrailingZeros(currentValue)
```

and you can't turn on formatting.